File: Programmers/2/154538. 숫자 변환하기/숫자 변환하기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

int solution(int x, int y, int n) {
    queue<pair<int, int>> q;
    vector<bool> visited(y + 1, false);
    
    q.push({x, 0});
    visited[x] = true;
    
    while (!q.empty())
    {
        auto [value, count] = q.front();
        q.pop();
        
        if (value == y) return count;
        
        int nexts[3] = {value + n, value * 2, value * 3};
        
        for (int next : nexts)
        {
            if (next <= y && !visited[next])
            {
                visited[next] = true;
                q.push({next, count + 1});
            }
        }
    }
    
    return -1;
}

File: Programmers/2/154538. 숫자 변환하기/숫자 변환하기.java
import java.util.*;

class Solution {
    public int solution(int x, int y, int n) {
        Queue<int[]> q = new LinkedList<>();
        boolean[] visited = new boolean[y + 1];
        
        q.add(new int[]{x, 0});
        visited[x] = true;
        
        while (!q.isEmpty()) {
            int[] cur = q.poll();
            int value = cur[0];
            int count = cur[1];
            
            if (value == y) return count;
            
            int[] nexts = {value + n, value * 2, value * 3};
            
            for (int next : nexts) {
                if (next <= y && !visited[next]) {
                    visited[next] = true;
                    q.add(new int[]{next, count + 1});
                }
            }
        }
        
        return -1;
    }
}